lib/src/Uncertainty/Algorithm/WeightedExperiments/ExperimentIntegration.cxx

```diff
@@ -0,0 +1,122 @@
+ *
+ */
+#include "openturns/OTprivate.hxx"
+#include "openturns/WeightedExperiment.hxx"
+#include "openturns/ExperimentIntegration.hxx"
+#include "openturns/Exception.hxx"
+#include "openturns/ResourceMap.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/SymbolicFunction.hxx"
+#include "openturns/DualLinearCombinationFunction.hxx"
+#include "openturns/ComposedFunction.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(ExperimentIntegration)
+
+typedef Collection<Function>                              FunctionCollection;
+
+/* Default constructor */
+ExperimentIntegration::ExperimentIntegration()
+  : PersistentObject()
+{
+  // Nothing to do
+}
+
+/* Constructor with parameters */
+ExperimentIntegration::ExperimentIntegration(const WeightedExperiment & weightedExperiment)
+  : PersistentObject()
+  , weightedExperiment_(weightedExperiment)
+{
+}
+
+/* Virtual constructor */
+ExperimentIntegration * ExperimentIntegration::clone() const
+{
+  return new ExperimentIntegration(*this);
+}
+
+/* String converter */
+String ExperimentIntegration::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << GetClassName()
+      << " name=" << getName ()
+      << " weightedExperiment=" << weightedExperiment_;
+  return oss;
+}
+
+Point ExperimentIntegration::integrate(const Function & g) const
+{
+  const UnsignedInteger outputDimension = g.getOutputDimension();
+  Point weights(0);
+  const Sample inputSample(weightedExperiment_.generateWithWeights(weights));
+  Point approximateIntegral(outputDimension);
+  const Sample outputSample(g(inputSample));
+  for (UnsignedInteger i = 0; i < outputDimension; ++i)
+  {
+    Point functionSampleMarginal(outputSample.getMarginal(i).asPoint());
+    approximateIntegral[i] = functionSampleMarginal.dot(weights);
+  }
+  return approximateIntegral;
+}
+
+Point ExperimentIntegration::computeL2Norm(const Function & g) const
+{
+  const UnsignedInteger outputDimension = g.getOutputDimension();
+  Point weights(0);
+  Sample inputSample(weightedExperiment_.generateWithWeights(weights));
+  if (g.getInputDimension() != inputSample.getDimension())
+    throw NotYetImplementedException(HERE) << "The input dimension of g is : " 
+                                           << g.getInputDimension() 
+                                           << " which is inconsistent with the experiment dimension : "
+                                           << inputSample.getDimension();
+  // Create the squared integrand
+  Description inputVariables(outputDimension);
+  Description formula(outputDimension);
+  for (UnsignedInteger i = 0; i < outputDimension; ++i)
+  {
+    const String variableName = (OSS() << "x" << i);
+    inputVariables[i] = variableName;
+    formula[i] = (OSS() << variableName << "^2");
+  }
+  SymbolicFunction squareFunction(inputVariables, formula);
+  Function integrand(ComposedFunction(squareFunction, g));
+  Point functionSquaredNorm(integrate(integrand));
```

Comment:
const

Reply:
Michaël, the following examples shows that the way you compute the conditional expectation is wrong in the dependent case (sorry the text is french...).

![PolyChaos_1](https://github.com/openturns/openturns/assets/13952415/18d309e5-f730-42f3-9996-5577216601e5)
![PolyChaos_2](https://github.com/openturns/openturns/assets/13952415/1cbe8831-fc6b-412a-837b-00e09e877d78)
![PolyChaos_3](https://github.com/openturns/openturns/assets/13952415/6cbcd694-86cf-4c1a-9c18-cf8c8773fede)
![PolyChaos_4](https://github.com/openturns/openturns/assets/13952415/e286f481-fa85-4a32-b6bf-7c7c93dcbb61)

Reply:
Fixed at bb5015f8a

Reply:
Anne's comment is unrelated to the ExperimentIntegration.cxx file. So I will take into account this particular comment in a specific commit.